s3/replication/manager/src/s3replicationmanager/manager_routes.py

```diff
@@ -0,0 +1,225 @@
+#
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from aiohttp import web
+from urllib.parse import urlparse, parse_qs
+import logging
+from s3replicationcommon.job import JobJsonEncoder
+import json
+
+
+LOG = logging.getLogger('manager')
+
+# Route table declaration
+routes = web.RouteTableDef()
+
+
+@routes.post('/subscribers')  # noqa: E302
+async def add_subscriber(request):
+    """Add subscriber
+
+    Handler for Subscriber
+    """
+    # Get subscriber
+    subscriber = await request.json()
+    LOG.debug('subscriber args are  : {}'.format(subscriber))
+
+    # Get subscriber id
+    sub_id = next(iter(subscriber))
+    subscriber_obj = request.app['subscribers']
+
+    # Check if subscriber is already present
+    if subscriber_obj.check_presence(sub_id):
+        return web.json_response(
+            {'Response': 'Replicator is Already subscribed!'})
+    else:
+        subscriber_obj.add_subscriber(subscriber)
+        LOG.debug(subscriber_obj.get_keys())
+        return web.json_response({'Response': 'subscriber added!'})
+
+
+@routes.get('/subscribers')  # noqa: E302
+async def list_subscribers(request):
+    """List subscriber
+
+    Handler to get subscriber list
+
+    """
+    return web.json_response({'subscribers': json.dumps(
+        list(request.app['subscribers'].get_keys()))}, status=200)
+
+
+@routes.delete('/subscribers/{sub_id}')  # noqa: E302
+async def remove_subscriber(request):
+    """Add subscriber
+
+    Handler for Subscriber
+
+    """
+    sub_id = (request.match_info['sub_id'])
+    # Check if subscriber is already present
+    if request.app['subscribers'].check_presence(sub_id):
+        request.app['subscribers'].remove_subscriber(sub_id)
+        return web.json_response(
+            {'Response': 'subscriber removed!'})
+    else:
+        return web.json_response({'ErrorResponse': 'subscriber not present!'})
+
+
+@routes.post('/jobs')  # noqa: E302
+async def add_job(request):
+    """Add jobs
+
+    Handler to add jobs to the job queue
+
+    """
+    job_record = await request.json()
+
+    # Get first key and find if already present
+    job_id = next(iter(job_record))
```

Comment:
Its a json dict right? you can just access job_record["id"] ?

Reply:
Yes, you are right. But, in my current implementation subscriber dict is taken as : e.g. : subscriber = {"subscriber1":{"foo": "bar"} }, not like subscriber = {"id":"subscriber1", "attr1" :  "x", "attr2": "y"}.

Reply:
I have added handling for second method for subscriber addition and handling in new commit.